### PR TITLE
DKIM Validator and Tools App Baseline (FAD-4223)

### PR DESCRIPTION
**Pull request by jasonrhodes**
If you grab this branch you can run it with `npm install` and then `npm start`.

If you want a test URL so you can see an email with pre-loaded results, try: 
http://localhost:3000/dkim/results/kdzrluvcg1f5sbq2@validator-uat.sparkpost.com

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,18 +1,99 @@
...
+    "chalk": "1.1.3",
+    "connect-history-api-fallback": "1.3.0",
+    "cross-spawn": "4.0.2",
+    "css-loader": "0.26.0",
+    "detect-port": "1.0.1",
+    "dotenv": "2.0.0",
+    "eslint": "3.8.1",
+    "eslint-config-react-app": "^0.5.0",
+    "eslint-loader": "1.6.0",
+    "eslint-plugin-flowtype": "2.21.0",
```
Comment (jonambas) [inline package.json]: We should probably go through these and remove what we don't need

File: package.json [hunk 2]
```diff
@@ -1,18 +1,99 @@
...
+    "babel-core": "6.17.0",
+    "babel-eslint": "7.1.1",
+    "babel-jest": "17.0.2",
+    "babel-loader": "6.2.7",
+    "babel-preset-react-app": "^2.0.1",
+    "case-sensitive-paths-webpack-plugin": "1.1.4",
+    "chalk": "1.1.3",
+    "connect-history-api-fallback": "1.3.0",
+    "cross-spawn": "4.0.2",
+    "css-loader": "0.26.0",
```
Comment (jonambas) [inline package.json]: I don't know if styles are a part of this ticket. If it is, we should replace `css-loader` with `sass-loader`

**Comment by jasonrhodes** [comment]
@jonambas yeah, on the one hand I agree with you. On the other hand, I was going to try to refactor down to a generic "Table" component but we have 3 tables and 1 of them is a totally different style and format than the other 2, so my decision was to keep it simple for now and then as we add new things, keep refactoring down into more generic component types. Curious to hear what others think, but I know I can get lost down a long rabbit hole of abstraction purity sometimes, and I'm trying to avoid that.

**Comment by jasonrhodes** [comment]
Totally on board with the overall structure though. We can move things into constants and routes.js, and otherwise I think we're on the same page.

**Comment by jonambas** [comment]
@jasonrhodes Yeah its tricky, maybe plug everything under a `dkim` folder for now, and we can pick stuff out if we find common patterns/markup

**Comment by jonambas** [comment]
@jasonrhodes take a look at this: https://github.com/JedWatson/classnames

```
var btnClass = classNames({
      'btn': true,
      'btn-pressed': this.state.isPressed,
      'btn-over': !this.state.isPressed && this.state.isHovered
    });
```

File: src/constants/text.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+export const DKIM_INTRO_TEXT = 'Verifying DKIM is one step that can help your email get to the inbox. Not only is it an industry standard, but when configured correctly it demonstrates to ISPs that – unlike spammers – you are accountable for your email. Check DKIM, quickly with this simple DKIM Validator tool.';
```
Comment (jonambas) [inline src/constants/text.js]: Yeah, I agree with this

Comment (jasonrhodes) [inline src/constants/text.js]: Sounds good to me, esp as we are going to be needing folders for components so the styles can sit next to them. (Sort of makes me wonder if tests should move back there too but leaving them in __tests__ for now)

File: src/components/App.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import Nav from './nav/Nav';
+import Footer from './footer/Footer';
+
+export default (props) => (
+  <div className='pageWrapper'>
+    <div className='container paddingTop--xxl paddingBottom--xxl'>
+      {props.children}
+    </div>
+    <Footer></Footer>
+    <Nav path={props.location.pathname}></Nav>
```
Comment (jonambas) [inline src/components/App.js]: Note to myself: props.location is global through react-router so no need to pass it down

File: src/utils.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+export function throttle(fn, threshhold = 250, context = this) {
```
Comment (jasonrhodes) [inline src/utils.js]: let's replace this with `_.throttle` from lodash, to lower what we have to maintain/think about here

https://lodash.com/docs/4.17.4#throttle

**Pull request merged**